File: chirp_control/lib/utils/units_repository.dart
import 'package:shared_preferences/shared_preferences.dart';

const _isMetricKey = 'units_is_metric';

Future<bool> loadIsMetric() async {
  final prefs = await SharedPreferences.getInstance();
  return prefs.getBool(_isMetricKey) ?? true;
}

Future<void> saveIsMetric(bool isMetric) async {
  final prefs = await SharedPreferences.getInstance();
  await prefs.setBool(_isMetricKey, isMetric);
}

double cmToDisplayUnit(double cm, bool isMetric) => isMetric ? cm : cm / 2.54;

String depthUnitLabel(bool isMetric) => isMetric ? 'cm' : 'in';

File: chirp-control-web/src/utils/unitsRepository.ts
const IS_METRIC_KEY = "units_is_metric";

export function loadIsMetric(): boolean {
  const raw = localStorage.getItem(IS_METRIC_KEY);
  return raw === null ? true : raw === "true";
}

export function saveIsMetric(isMetric: boolean): void {
  localStorage.setItem(IS_METRIC_KEY, String(isMetric));
}

export function cmToDisplayUnit(cm: number, isMetric: boolean): number {
  return isMetric ? cm : cm / 2.54;
}

export function depthUnitLabel(isMetric: boolean): string {
  return isMetric ? "cm" : "in";
}
